offscreen_effect_player/yuv_converter.cpp

```diff
@@ -0,0 +1,359 @@
+#include "yuv_converter.hpp"
+#include <string>
+#include <opengl.hpp>
+
+static const char* to_gl_check_framebuffer_status(GLenum e)
+{
+    switch(e) {
+        case GL_FRAMEBUFFER_UNDEFINED:
+            return "GL_FRAMEBUFFER_UNDEFINED";
+
+        case GL_FRAMEBUFFER_INCOMPLETE_ATTACHMENT:
+            return "GL_FRAMEBUFFER_INCOMPLETE_ATTACHMENT";
+
+        case GL_FRAMEBUFFER_INCOMPLETE_MISSING_ATTACHMENT:
+            return "GL_FRAMEBUFFER_INCOMPLETE_MISSING_ATTACHMENT";
+
+        case GL_FRAMEBUFFER_UNSUPPORTED:
+            return "GL_FRAMEBUFFER_UNSUPPORTED";
+
+        case GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE:
+            return "GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE";
+
+    }
+    return "UNKNOWN GLenum";
+}
+
+static void put_error_message(const char* where, const char* what)
+{
+    std::string msg;
+    if (where) {
+        msg.append(where);
+        msg.append("\n");
+    }
+    if (what) {
+        msg.append(what);
+        msg.append("\n");
+    }
+    throw std::runtime_error(msg);
+}
+
+namespace bnb::converter
+{
+
+const int drawing_plane_vert_count = 4;
+
+#define N "\n"

```

Comment:
don't like such defines, I don't think that it is worth to save only one character